integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/load/Compactor.scala

```diff
@@ -121,10 +124,17 @@ object Compactor {
           segmentIdToLoadStartTimeMapping(validSegments.head),
           SegmentStatus.SUCCESS,
           carbonLoadModelForMergeDataFiles.getFactTimeStamp, rebuiltSegments.toList.asJava)
-
+        siCompactionIndexList ::= indexCarbonTable
       } catch {
         case ex: Exception =>
           LOGGER.error(s"Compaction failed for SI table ${secondaryIndex.indexName}", ex)
+          siCompactionIndexList.foreach { indexCarbonTable =>
```

Comment:
please add a comment here, what you are doing in which scenario

Reply:
OK